src/renderer/services/chain/fees/stake.ts

```diff
@@ -67,28 +94,32 @@ const stakeFeeByChain$ = (chain: Chain): FeeLD => {
   }
 }
 
-const stakeFees$: StakeFeesLD = selectedPoolChain$.pipe(
-  RxOp.switchMap((oPoolChain) =>
-    FP.pipe(
-      oPoolChain,
-      O.map((chain) =>
-        FP.pipe(
-          Rx.combineLatest(
-            isBaseChain(chain)
-              ? // for deposits on base chain, fee for base chain is needed only
-                [stakeFeeByChain$(BASE_CHAIN)]
-              : // for x-chain deposits, we do need to load fees for base- AND x-chain,
-                [stakeFeeByChain$(BASE_CHAIN), stakeFeeByChain$(chain)]
-          ),
-          RxOp.map(sequenceTRDFromArray),
-          liveData.map(([base, cross]) => ({
-            base,
-            cross: O.fromNullable(cross)
-          }))
-        )
-      ),
-      O.getOrElse((): StakeFeesLD => Rx.of(RD.initial))
+// TODO (@Veado) Store results of deposit txs into a state, so views will have access to it.
```

Comment:
"Store results of deposit `txs`(why txs ? maybe fees ?) into a state, so views will have access to it."

if yes we can just create new `observableState" for type
```ts
const {get$: type$, set: setType} = createObservableState()
```
and after just use it as
```ts
const stakeFees$: StakeFeesLD = combineLatest(selectedPoolChain$, type$) // ... and all prev logic here
```
and at the `AddStakeView.tsx`
just 
```ts
useEffect(() => setType(type))
```

what do you think ?

Reply:
Yep, something like this. We will implement it later while finishing all needed stuff for deposit txs #537